<launch>

  <!-- Hybrid A* node -->

  <!-- common configurations -->
  <arg name="costmap_topic" default="global_cost_map" />
  <arg name="waypoints_velocity" default="5.0" />
  <arg name="update_rate" default="10" />
  <arg name="is_visual" default="false" />

  <!-- Hybrid A* search configurations -->
  <arg name="time_limit" default="10000.0" />

  <arg name="vehicle_length" default="4.788" />
  <arg name="vehicle_width" default="2.198" />
  <arg name="vehicle_cg2back" default="1.367" />

  <arg name="max_turning_radius" default="20.0" />
  <arg name="min_turning_radius" default="4.36" />
  <arg name="turning_radius_size" default="11" />
  <arg name="theta_size" default="72" />

  <arg name="reverse_weight" default="2.0" />
  <arg name="turning_weight" default="1.2" />

  <arg name="goal_lateral_tolerance" default="1.0" />
  <arg name="goal_longitudinal_tolerance" default="1.0" />
  <!-- 3Â° -->
  <arg name="goal_angular_tolerance" default="0.15708" />
  <arg name="obstacle_threshold" default="100" />

  <arg name="use_back" default="true" />
  <arg name="use_reeds_shepp" default="true" />

  <node pkg="hp_hybrid_astr_node" type="hp_hybrid_astr_node" name="hp_hybrid_astr_node" output="screen">

    <param name="waypoints_velocity" value="$(arg waypoints_velocity)" />
    <param name="update_rate" value="$(arg update_rate)" />
    <param name="is_visual" value="$(arg is_visual)" />
    <param name="costmap_topic" value="$(arg costmap_topic)"/>

    <param name="time_limit" value="$(arg time_limit)" />
    <param name="vehicle_length" value="$(arg vehicle_length)" />
    <param name="vehicle_width" value="$(arg vehicle_width)" />
    <param name="vehicle_cg2back" value="$(arg vehicle_cg2back)" />
    <param name="max_turning_radius" value="$(arg max_turning_radius)" />
    <param name="min_turning_radius" value="$(arg min_turning_radius)" />
    <param name="turning_radius_size" value="$(arg turning_radius_size)" />
    <param name="theta_size" value="$(arg theta_size)" />
    <param name="reverse_weight" value="$(arg reverse_weight)" />
    <param name="turning_weight" value="$(arg turning_weight)" />
    <param name="goal_lateral_tolerance" value="$(arg goal_lateral_tolerance)" />
    <param name="goal_longitudinal_tolerance" value="$(arg goal_longitudinal_tolerance)" />
    <param name="goal_angular_tolerance" value="$(arg goal_angular_tolerance)" />
    <param name="obstacle_threshold" value="$(arg obstacle_threshold)" />
    <param name="use_back" value="$(arg use_back)" />
    <param name="use_reeds_shepp" value="$(arg use_reeds_shepp)" />
  </node>

</launch>
